src/server/game/AI/ScriptedAI/ScriptedCreature.h

```diff
@@ -210,11 +210,16 @@ struct TC_GAME_API ScriptedAI : public CreatureAI
     //Plays a sound to all nearby players
     void DoPlaySoundToSet(WorldObject* source, uint32 soundId);
 
-    //Drops all threat to 0%. Does not remove players from the threat list
-    void DoResetThreat();
-
-    float DoGetThreat(Unit* unit);
-    void DoModifyThreatPercent(Unit* unit, int32 pct);
+    // Add specified amount of threat directly to victim (ignores redirection effects) - also puts victim in combat and engages them if necessary
+    void AddThreat(Unit* victim, float amount, Unit* who = nullptr) { (who ? who : me)->AddThreat(victim, amount, nullptr, true, true); }
```

Comment:
Alright, I cede that - however, I have to admit I still don't understand the thought process behind not wanting to "have to" include `Creature.h` in `ScriptedCreature.h`.
What use case do you see where a script would include the latter without also wanting to include the former? Have you ever seen a script that doesn't use any of `Creature`'s defined methods directly?

If the answer to that is "no" (and I strongly believe it to be), I don't see any harm in leaving these methods as inline definitions - though the comment on the include should be amended in that case, of course.